cpp/pgen2.cc

```diff
@@ -2,14 +2,23 @@
 
 #include "pgen2.h"
 
-namespace parse {
+#include <vector>
 
-void Parser::setup(int start) {
-  assert(0);
-}
+namespace pnode {
+
+static std::vector<PNode> arena;
 
-bool Parser::addtoken(int typ, syntax_asdl::Token* opaque, int ilabel) {
-  assert(0);
+PNode* NewPNode(int typ, syntax_asdl::Token* tok,
+                List<pnode::PNode*>* children) {
+  // This is only ever called with a null token from Parser::setup(). Init/reset
+  // the PNode arena in that case.
+  if (tok == nullptr) {
+    arena.reserve(1000000);
+    arena.clear();
+  }
+  DCHECK(arena.size() < arena.capacity());
```

Comment:
This should be CHECK() until we make it dynamically sized, i.e. always on


I learned the hard way by running out of GC objects :)

Reply:
Oof good catch